lib/test.js

```diff
@@ -240,11 +247,11 @@ Test.prototype._publicApi = function () {
 	}
 
 	// Patched assert methods: increase assert count and store errors.
-	Object.keys(assert).forEach(function (el) {
+	Object.keys(Assert.prototype).forEach(function (el) {
 		api.skip[el] = skipFn;
 		api[el] = function () {
 			try {
-				var fn = assert[el].apply(assert, arguments);
+				var fn = self.assert[el].apply(self.assert, arguments);
```

Comment:
> Do we have to make assert a class? Can't we just track assertions here?

I'm open to suggestions!

> Demanding the assert class track assertions is going to make it harder to support custom assertions down the road.

It will, see my comment below.

> Is the issue simply that it is hard to know which argument is the message here?

The issue lies in AVA's feature - concurrency. As you know, tests run concurrently, so we have to track which assertions happened during that particular test. That's why `Assert` class is needed, because that way we can safely get a list of test's assertions.